Update documentation and improve code formatting

jacobwhall opened the pull request:
- code reformatting
- README.md overhaul

jacobwhall committed:
cleaning up src/dhs_availability a bit

- adding and removing imports
- using os.path.join() for safer path joins (more can be done here)
- making quotation makes consistent ("" vs. '')

jacobwhall committed:
reformatting src/dhs_clusters.py

- consistent quotation marks ("" vs. '')
- expanding some statements out for readability

jacobwhall committed:
update README.md

- added information about registering to download DHS data
  above in-depth instructions
- broke instructions out into smaller sections:
  - Setting up your environment
  - Downloading data
  - Setting configuration options
  - Processing data and training models
- various formatting, wording, spelling adjustments
- renamed LICENSE to LICENCE.md because its written in markdown
- linking to LICENSE.md from README.md

jacobwhall committed:
load MLflow URI from config.ini in src/models.py

jacobwhall committed:
Merge branch 'aiddata:prefect-update' into prefect-update

sgoodm commented on README.md:
````diff
@@ -1,93 +1,66 @@
 		```
 		conda env export > environment.yml
 		```
-    - Add the `src` directory to your Conda environment Python path:
````
@jacobwhall This step is needed to enable importing the model_utils.py file

sgoodm commented on README.md:
```diff
@@ -1,93 +1,66 @@
 2. Create Conda environment
 
-	- To create a new environment:
-        - From scratch
```
Hesitant to drop the "build from scratch" step here entirely as there's a this would likely fail as soon as someone tries to build with a different python version

sgoodm commented on README.md:
```diff
@@ -1,93 +1,66 @@
 3. Install SpatiaLite 5.1.0+
     - We use [VirtualKNN2](https://www.gaia-gis.it/fossil/libspatialite/wiki?name=KNN2), which was added in [check-in 03786a62cd](https://www.gaia-gis.it/fossil/libspatialite/info/03786a62cdb4ab17) of SpatialLite.
-    - Download and build Spatialite:
+    - Download and build SpatiaLite:
```
To be replaced with QGIS install and separate SpatialLite readme as discussed

jacobwhall committed:
add prefect to conda environment

jacobwhall committed:
squash some bugs in get_dhs_adm_units.py

jacobwhall committed:
merge remote branch

jacobwhall committed:
remove environment prefixes from environment.yml, update command

- also used verbose option names in the conda commands we have in
  README.md to make them more reader-friendly

jacobwhall committed:
add instructions for building env from scratch back into README.md

jacobwhall committed:
recommend installing QGIS in README.md

- move libspatialite installation instructions to
  docs/install_libspatialite.md

jacobwhall committed:
put conda `src` dir instructions back in README.md
